### ISSUE TITLE
Helm: Don't rely on .Values.image.tag for version compatibility

### ISSUE THREAD

**Issue by mrueg** [coreos/prometheus-operator#1233]
**What did you do?**
I used my custom prometheus containers with its own versioning
**What did you expect to see?**
I expected to see the right api choices e.g. in https://github.com/coreos/prometheus-operator/blob/c96fa34640fbf7584cf3fc54f4fbfc96dc38b502/helm/prometheus-operator/templates/deployment.yaml#L30
**What did you see instead? Under which circumstances?**
It picked the wrong path, as the schema is YYYYMMDD.

Would it be possible to define a .Values.image.compat that can be used instead?

**Comment by gianrubio** [comment]
@mrueg I don't think we should support version < 13.0 so I don't mind to remove this validation, would you like to contribute to the project making a PR to fix this issue?

**Comment by mrueg** [comment]
Sure I can provide a PR. There might be other occasions as well.

**Comment by brancz** [comment]
Note that the Prometheus images used are required to be equivalent to the upstream images and tags must be the same as well, as the it is used to parse semantic versions to make decisions to generate things in backward compatible ways.
